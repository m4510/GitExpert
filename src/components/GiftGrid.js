import React from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifts } from '../hooks/useFetchGifts';

export const GiftGrid = ({ category }) => {
  const { loading, data: images } = useFetchGifts(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && 'Loading...'}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
