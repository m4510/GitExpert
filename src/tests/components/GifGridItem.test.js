import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Testing GifGridItem', () => {
  const title = 'Some Title';
  const url = 'http://localhost/some.png';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should render component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
