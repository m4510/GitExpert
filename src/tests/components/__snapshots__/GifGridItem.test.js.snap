// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem should render component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Some Title"
    src="http://localhost/some.png"
  />
  <p>
    Some Title
  </p>
</div>
`;
